der/derive/src/sequence.rs

```diff
@@ -127,35 +130,39 @@ impl SequenceField {
         });
 
         let attrs = FieldAttrs::parse(&field.attrs, type_attrs);
+        let field_type = field.ty.clone();
 
         if attrs.tag_mode == TagMode::Implicit {
             abort!(ident, "IMPLICIT tagging not supported for `Sequence`");
         }
 
-        Self { ident, attrs }
+        Self {
+            ident,
+            attrs,
+            field_type,
+        }
     }
 
     /// Derive code for decoding a field of a sequence.
     fn to_decode_tokens(&self) -> TokenStream {
         let ident = &self.ident;
-        if self.attrs.asn1_type.is_some() {
-            let field_decoder = self.attrs.decoder();
-            quote! { let #ident = #field_decoder.try_into()?; }
+        let ty = self.field_type.clone();
+        if let Some(default) = &self.attrs.default {
+            quote!(let mut #ident = Some(decoder.decode::<#ty>()?.unwrap_or_else(#default));)
```

Comment:
Thought I'd merge this and try to clean up my remaining nits, but this seems a bit problematic. I'm not sure this is what you actually intended?

It seems like when `default` is present, it should attempt to decode the given type as `Option<#ty>`, and then call `unwrap_or_default` on the result, if the `Option` decodes successfully.

Reply:
OK. I think the Option<> on the default'ed fields will likely go away as Context Specific is worked. The goal here was to *always* return a value where there was a default, perhaps stretching Option a bit.

Reply:
I see now. The Option<> became unnecessary as default support was added. Originally, the absence of the field signified default and the caller had to supply the default value. Removing Option was natural next step.